File: isma-intg-lib/isma-intg-lib-rk3/src/main/java/ru/nstu/isma/intg/lib/rungeKutta/rk3/internal/Rk3Stage2Calculator.java
package ru.nstu.isma.intg.lib.rungeKutta.rk3.internal;

import ru.nstu.isma.intg.api.methods.StageCalculator;

/**
 * @author Mariya Nasyrova
 * @since 04.10.14
 */
public class Rk3Stage2Calculator extends StageCalculator {

    @Override
    public double yk(double step, double y, double f, double[] stages) {
        return y + 0.5 * stages[0];
    }

    @Override
    public double k(double step, double y, double f, double[] stages, double stagesY, double stagesF) {
        return step * stagesF;
    }

}

File: isma-intg-lib/isma-intg-lib-rk3/src/main/kotlin/ru/nstu/isma/intg/lib/rungeKutta/rk3/internal/Rk3Stage2Calculator.kt
package ru.nstu.isma.intg.lib.rungeKutta.rk3.internal

import ru.nstu.isma.intg.api.methods.StageCalculator

/**
 * @author Mariya Nasyrova
 * @since 04.10.14
 */
class Rk3Stage2Calculator : StageCalculator() {
    override fun yk(step: Double, y: Double, f: Double, stages: DoubleArray): Double {
        return y + 0.5 * stages[0]
    }

    override fun k(step: Double, y: Double, f: Double, stages: DoubleArray, stagesY: Double, stagesF: Double): Double {
        return step * stagesF
    }
}
